cli/src/services/session.service.ts

```diff
@@ -0,0 +1,81 @@
+import fs from 'node:fs';
+import yaml from 'yaml';
+import path from 'node:path';
+import { ImmichApi } from '../api/client';
+import { LoginError } from '../cores/errors/login-error';
+
+export class SessionService {
+  readonly configDir: string;
+  readonly authPath!: string;
+  private api!: ImmichApi;
+
+  constructor(configDir: string) {
+    this.configDir = configDir;
+    this.authPath = path.join(this.configDir, 'auth.yml');
+  }
+
+  public async connect(): Promise<ImmichApi> {
+    await fs.promises.access(this.authPath, fs.constants.F_OK).catch((error) => {
+      if (error.code === 'ENOENT') {
+        throw new LoginError('No auth file exist. Please login first');
+      }
+    });
+
+    const data: string = await fs.promises.readFile(this.authPath, 'utf8');
+    const parsedConfig = yaml.parse(data);
+    const instanceUrl: string = parsedConfig.instanceUrl;
+    const apiKey: string = parsedConfig.apiKey;
+
+    if (!instanceUrl) {
+      throw new LoginError('Instance URL missing in auth config file ' + this.authPath);
+    }
+
+    if (!apiKey) {
+      throw new LoginError('API key missing in auth config file ' + this.authPath);
+    }
+
+    this.api = new ImmichApi(instanceUrl, apiKey);
+
+    await this.ping();
+
+    return this.api;
+  }
+
+  public async keyLogin(instanceUrl: string, apiKey: string): Promise<ImmichApi> {
+    this.api = new ImmichApi(instanceUrl, apiKey);
+
+    // Check if server and api key are valid
+    const { data: userInfo } = await this.api.userApi.getMyUserInfo().catch((error) => {
+      throw new LoginError(`Failed to connect to the server: ${error.message}`);
+    });
+
+    console.log(`Logged in as ${userInfo.email}`);
+
+    if (!fs.existsSync(this.configDir)) {
+      // Create config folder if it doesn't exist
+      fs.mkdirSync(this.configDir, { recursive: true });
+    }
+
+    fs.writeFileSync(this.authPath, yaml.stringify({ instanceUrl: instanceUrl, apiKey: apiKey }));
```

Comment:
If the key and value are the same you can just do this `({ instanceUrl, apiKey })`

Reply:
Done